service/history/execution/context.go

```diff
@@ -265,11 +270,27 @@ func NewContext(
 }
 
 func (c *contextImpl) Lock(ctx context.Context) error {
-	return c.mutex.Lock(ctx)
+	err := c.mutex.Lock(ctx)
+	if err != nil {
+		return err
+	}
+	c.lockTime.Store(time.Now().UnixNano())
```

Comment:
no need to blend atomics and mutexes, it's likely to perform worse

Reply:
I didn't add for perf reasons actually. Swap sounded convenient :) However I will move things to critical section to make it race free